core/src/main/java/com/microsoft/applicationinsights/TelemetryConfiguration.java

```diff
@@ -108,15 +109,18 @@ public static TelemetryConfiguration createDefault() {
      * Gets the telemetry channel.
      * @return An instance of {@link com.microsoft.applicationinsights.channel.TelemetryChannel}
      */
-    public TelemetryChannel getChannel() {
+    public synchronized TelemetryChannel getChannel() {
```

Comment:
Is NopTelemetryChannel.instance() thread-safe? If not, then there is a theoretical chance of a mishap, especially if it's lazy initialization.

Reply:
I added it because `setChannel` is exposed. If the user intends to provide their own channel programmatically, I wanted to be sure the write completes if there's a parallel `getChannel` call.

but I could be convinced to remove it.

NopTelemetryChannel.instance() is thread-safe.